dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorArm8.java

```diff
@@ -171,7 +171,7 @@ public List<Event> visitLlvmCmpXchg(LlvmCmpXchg e) {
     @Override
     public List<Event> visitLlvmFence(LlvmFence e) {
         String mo = e.getMo();
-        Fence fence = mo.equals(C11.MO_RELEASE) || mo.equals(C11.MO_ACQUIRE_RELEASE) || mo.equals(C11.MO_SC) ?
+        GenericVisibleEvent fence = mo.equals(C11.MO_RELEASE) || mo.equals(C11.MO_ACQUIRE_RELEASE) || mo.equals(C11.MO_SC) ?
```

Comment:
I would rather use `Event` as variable type. The code doesn't really need to know about what type of event it is. I think we can do that for pretty much all cases where `GenericVisibleEvent` is used right now.

Reply:
What about the factory? There we always return the most precise type so for consistency I would still use `GenericVisibleEvent`

Reply:
In the factory you can use `GenericVisibleEvent`. I would see it similar to how we use `Set<T>` as a type even if we instantiate it with `HashMap<T>` most of the time. If the concrete type is not so important at the call site, I would stick with the more generic type there.